enterprise/audit/table.go

```diff
@@ -149,17 +165,50 @@ var AuditableResources = auditMap(map[any]map[string]Action{
 		"ip_address":       ActionIgnore,
 		"scope":            ActionIgnore,
 		"token_name":       ActionIgnore,
-	},
+	})).
 	// TODO: track an ID here when the below ticket is completed:
 	// https://github.com/coder/coder/pull/6012
-	&database.License{}: {
+	Add(entry(database.License{}, map[string]Action{
 		"id":          ActionIgnore,
 		"uploaded_at": ActionTrack,
 		"jwt":         ActionIgnore,
 		"exp":         ActionTrack,
 		"uuid":        ActionTrack,
-	},
-})
+	}))
+
+// entry is a helper function that ensures all entries in the table are valid
+// audit.Auditable types. It also ensures all json tags have a corresponding
+// action.
+func entry[A audit.Auditable](v A, f map[string]Action) (string, map[string]Action) {
```

Comment:
Can we merge `entry` and `add` into one? That way it's not doubly nested

Reply:
I am trying to play with the syntax still. This code does what I want, it checks enterprise matches APGL (but not the other way around...).


You cannot have a generic type constraint on a method (func attached to struct). So I have to make the `entry` function outside the `Table`. 

All generic type constraints are not available at runtime, so I cannot use `reflect` on `audit.Auditable`. I _can use `"golang.org/x/tools/go/packages"`_ and parse the APGL `audit` package.... which I might have to do to make sure the generic type constraint is fully represented in the table :cry:.

----

I am going to spend a little more time and see if I can make this a bit better syntax wise.